jupyter_accessibility/dodo.py

```diff
@@ -46,87 +33,75 @@ def remove_directory(*dir):
-
-    It initializes the project and invokes the primary doit application."""
-
-    ids: list[str]
-    dir: Path = Path("jupyter-a11y-build")
-    project: type = None
-    app: object = None
-
-    def __post_init__(self):
-        self.setup_project()
-
-    def setup_project(self):
-        from .project import Project
-
-        self.project = Project(**self.dict(exclude={"project", "app"}))
-
-    def main(self, args=None, standalone=False):
-        """invoke the doit application through an interactive api."""
-        try:
-            self.project.doit().run(prep_args(args))
-        except standalone and SystemExit or () as e:
-            if e.args[0] != 0:
-                raise e
-
-    @classmethod
-    def run(cls, args=None, standalone=False):
-        """invoke the doit application as a standalone command line tool."""
-        # parse the argument we know and pass the rest to the doit application.
-        ns, args = cls.parse_args(args)
-        main = cls(**ns.__dict__)
-
-        main.main(args, standalone=standalone)
-
-    @classmethod
-    def get_parser(cls):
-        """the argument parser for the Main cli"""
-        from argparse import ArgumentParser
-
-        parser = ArgumentParser("builder")
-        parser.add_argument("-i", "--ids", default=["jupyterlab"], help="repository ids", nargs="*")
-        parser.add_argument(
-            "-d", "--dir", default=Path("jupyter-ally-build"), help="the build directory", type=Path
-        )
-        return parser
-
-    @classmethod
-    def parse_args(cls, args=None):
-        parser = cls.get_parser()
-        return parser.parse_known_args(args=prep_args(args))
-
-
-class Tasks(Base):
-    def doit(self):
-        """build the doit application by pulling tasks from the"""
-        from doit.cmd_base import ModuleTaskLoader
-        from doit.doit_cmd import DoitMain
-
-        tasks = dict((x, getattr(self, x)) for x in dir(self) if x.startswith("task_"))
-        tasks.update({"DOIT_CONFIG": DOIT_CONFIG})
-        return DoitMain(ModuleTaskLoader(tasks))
-
-    def commands(self):
-        """explore what doit knows"""
-        return self.doit().get_commands()
-
-    def do(self, x, **kwargs):
-        """create a cmd action that can be run from the api"""
-        return do(f"{self.conda} {x}", **kwargs)
-
-
-def prep_args(x):
-    if isinstance(x, str):
-        from shlex import split
-
-        return split(x)
-    return x
+##############################################################################################
+## Tasks
+##############################################################################################
+
+
+build_dir = ROOT / "build"
+conda_env_dir = build_dir / ".env"
+prefix = conda_env_dir.resolve()
+conda = f"conda run --no-capture-output --prefix {prefix}"
+
+def task_create_env():
+    """Create a test environment into which dependencies needed by the test code will be installed"""
+    yield dict(
```

Comment:
Why are you using `yield` instead of `return`?

Reply:
Copy-paste from old code. ;)

In the next iteration of this PR, before I take it out of draft mode, I'm going to address this.